src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc

```diff
@@ -0,0 +1,401 @@
+                   &executable_response->version);
+  auto executable_output_it = executable_output->object().find("success");
+  if (!isKeyPresent(executable_output, "success")) {
+    *error = GRPC_ERROR_CREATE(
+        "The executable response must contain the "
+        "`success` field.");
+    return nullptr;
+  }
+  executable_response->success = executable_output_it->second.boolean();
+  if (executable_response->success) {
+    if (!isKeyPresent(executable_output, "token_type")) {
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain the "
+          "`token_type` field.");
+      return nullptr;
+    }
+    executable_response->token_type =
+        gpr_strdup(getStringValue(executable_output, "token_type").c_str());
+    executable_response->expiration_time =
+        gpr_time_to_millis(gpr_inf_future(GPR_CLOCK_REALTIME));
+    if (output_file_path_ != "" &&
+        !isKeyPresent(executable_output, "expiration_time")) {
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain the `expiration_time` field "
+          "for successful responses when an output_file has been specified in "
+          "the configuration.");
+      return nullptr;
+    }
+    if (isKeyPresent(executable_output, "expiration_time")) {
+      if (!absl::SimpleAtoi(
+              getStringValue(executable_output, "expiration_time"),
+              &executable_response->expiration_time)) {
+        *error = GRPC_ERROR_CREATE(
+            "The executable response contains an invalid value for "
+            "`expiration_time`.");
+        return nullptr;
+      }
+    }
+    if (strcmp(executable_response->token_type, SAML_SUBJECT_TOKEN_TYPE) == 0)
+      executable_output_it = executable_output->object().find("saml_response");
+    else
+      executable_output_it = executable_output->object().find("id_token");
+    if (executable_output_it == executable_output->object().end()) {
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain a valid token.");
+      return nullptr;
+    }
+    executable_response->subject_token =
+        gpr_strdup(executable_output_it->second.string().c_str());
+  } else {
+    if (!isKeyPresent(executable_output, "code")) {
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain the "
+          "`code` field when unsuccessful.");
+      return nullptr;
+    }
+    executable_response->error_code =
+        gpr_strdup(getStringValue(executable_output, "code").c_str());
+    if (!isKeyPresent(executable_output, "message")) {
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain the "
+          "`message` field when unsuccessful.");
+      return nullptr;
+    }
+    executable_response->error_message =
+        gpr_strdup(getStringValue(executable_output, "message").c_str());
+  }
+  return executable_response;
+}
+
+void PluggableAuthExternalAccountCredentials::RetrieveSubjectToken(
```

Comment:
This function is pretty long. I would split it up into ~3 (validate env, retrieve from outputfile, call executable)